Tests/Integration/BehaviorTests/tests/type_conversions/test/test/test.js

```diff
@@ -0,0 +1,48 @@
+// RUN: cd %S && truffle test
+
+var config = require("../config.js")
+
+var Contract = artifacts.require("./" + config.contractName + ".sol");
+var Interface = artifacts.require("./_Interface" + config.contractName + ".sol");
+Contract.abi = Interface.abi
+
+var ExternalContract = artifacts.require("./External.sol");
+
+contract(config.contractName, function() {
+  it("should be possible to cast from a lower bit type to a higher bit type", async function() {
+    const external = await ExternalContract.deployed();
+    const instance = await Contract.deployed();
+
+    let t;
+    t = await instance.testReduce(100, external.address);
+    assert.equal(t.valueOf(), 100);
+  });
+
+  it("should be possible to cast from a higher bit type to a lower bit type for small values", async function() {
+    const external = await ExternalContract.deployed();
+    const instance = await Contract.deployed();
+
+    let t;
+    const value = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF / 2;
```

Comment:
No, these cannot work. Anything outside 64-bit is automatically a floating-point number in Javascript.

```
$ node
> 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
3.402823669209385e+38
> 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF - 1
3.402823669209385e+38
> 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF == (0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF - 1)
true
```

Although documentation seems scarce, [this link](https://ethereum.stackexchange.com/questions/29597/how-to-convert-bignumber-to-number-in-truffle-framework) (the question actually) seems to suggest you can use string to represent call arguments with large values.

Reply:
I wasn't sure about this. Thanks so much! The documentation is seriously lacking… I tried finding an API doc for truffle but couldn't even find that. Let's use a string then!